canonical-web-and-design/hiring.canonical.com
# Use dblink to send SQL requests

# Summary

## The problem 

Due to the lack of access that we can get from our K8s cluster to the redshift database we need to find alternative solutions to access it.

One solution is to give access to redshift to our own database a run the query from it. This is [documented in this article](https://aws.amazon.com/blogs/big-data/join-amazon-redshift-and-amazon-rds-postgresql-with-dblink/) from AWS.

This PR refactors the code to now run all queries through an external database. The database can be local or distant. Until [this ticket](https://portal.admin.canonical.com/C133921/) is not done we will need to use a local database for dev.

Fixes https://github.com/canonical-web-and-design/workplace-engineering-squad/issues/157

## Done

- Refactor code to run all SQL queries through an external database
- Add setup and reset scripts
- Update Readme

# QA

- Follow the steps in the README
- From those steps you should be able to setup your database and start the service
- Once it started, with multiple users access pages like: http://0.0.0.0:8501/stages/application-review
- Make sure the data makes sense :smile:
